### PR TITLE
CEDS-2251 - Skip 'Other Parties' page for Clearance Requests

### PR DESCRIPTION
**Pull request by iservport**
Reintroduced.

For Clearance Requests, the '3/37 Additional Supply Chain Actors' data elements are not required in any circumstances. It is currently included as an optional field and it is not relevant to Clearance Requests.

### CODE REVIEW

File: test/unit/controllers/declaration/DeclarationAdditionalActorsControllerSpec.scala [hunk 1]
```diff
@@ -164,6 +164,17 @@ class DeclarationAdditionalActorsControllerSpec extends ControllerSpec with Erro
 
     "return 303 (SEE_OTHER)" when {
 
+      onJourney(CLEARANCE)() { request =>
+        "redirect 303 (See Other) to start" in new SetUp {
```
Comment (bambuchaAdm) [inline test/unit/controllers/declaration/DeclarationAdditionalActorsControllerSpec.scala]: Wording in this test looks little silly
``` return 303 (SEE_OTHER) when on clearance journey redirect 303 (See Other) to Start``` - is here you mean "back to journey start" ?

File: app/controllers/declaration/RepresentativeDetailsController.scala [hunk 2]
```diff
@@ -64,8 +64,10 @@ class RepresentativeDetailsController @Inject()(
     declarationType match {
       case DeclarationType.SUPPLEMENTARY =>
         controllers.declaration.routes.DeclarationAdditionalActorsController.displayPage
-      case DeclarationType.STANDARD | DeclarationType.SIMPLIFIED | DeclarationType.OCCASIONAL | DeclarationType.CLEARANCE =>
+      case DeclarationType.STANDARD | DeclarationType.SIMPLIFIED | DeclarationType.OCCASIONAL =>
         controllers.declaration.routes.CarrierDetailsController.displayPage
+      case DeclarationType.CLEARANCE =>
+        controllers.declaration.routes.DeclarationHolderController.displayPage
```
Comment (janeturner) [inline app/controllers/declaration/RepresentativeDetailsController.scala]: Representative details should lead to carrier details now on clearance (this was part of my ticket 2245)

File: app/controllers/declaration/CarrierDetailsController.scala [hunk 3]
```diff
@@ -61,11 +62,19 @@ class CarrierDetailsController @Inject()(
...
     }
 
+  private def nextPage(declarationType: DeclarationType): Mode => Call =
+    declarationType match {
+      case DeclarationType.CLEARANCE =>
+        controllers.declaration.routes.DeclarationHolderController.displayPage
+      case _ =>
+        controllers.declaration.routes.DeclarationAdditionalActorsController.displayPage
+    }
+
```
Comment (SteveSugden) [inline app/controllers/declaration/CarrierDetailsController.scala]: 2251 - additional actors is skipped for Clearance

File: app/forms/declaration/CarrierDetails.scala [hunk 4]
```diff
@@ -30,10 +30,10 @@ object CarrierDetails extends DeclarationPage {
   val id = "CarrierDetails"
 
   val mappingEitherRequired = Forms.mapping("details" -> EntityDetails.defaultMapping)(CarrierDetails.apply)(CarrierDetails.unapply)
-  val mappingOptional = Forms.mapping("details" -> EntityDetails.mappingOptional)(CarrierDetails.apply)(CarrierDetails.unapply)
+  val optionalMapping = Forms.mapping("details" -> EntityDetails.optionalMapping)(CarrierDetails.apply)(CarrierDetails.unapply)
 
   def form(declarationType: DeclarationType): Form[CarrierDetails] = declarationType match {
-    case CLEARANCE => Form(mappingOptional)
+    case CLEARANCE => Form(optionalMapping)
```
Comment (SteveSugden) [inline app/forms/declaration/CarrierDetails.scala]: 👍 2245 carrier details optional for clearance

**Pull request merged**